server/src/main/java/org/cloudfoundry/identity/uaa/security/ScimUserUpdateDiff.java

```diff
@@ -40,7 +39,8 @@ public boolean isAnythingOtherThanNameDifferent(String userId, ScimUser scimUser
 
     private boolean originsEquivalent(ScimUser scimUserFromRequest, ScimUser scimUserFromDb) {
         return scimUserFromDb.getOrigin().equals(scimUserFromRequest.getOrigin()) ||
-                (scimUserFromDb.getOrigin().equals(OriginKeys.UAA) && Strings.isEmpty(scimUserFromRequest.getOrigin()));
+                (scimUserFromDb.getOrigin().equals(OriginKeys.UAA) && (scimUserFromRequest.getOrigin() == null || scimUserFromRequest.getOrigin()
+                        .isEmpty()));
```

Comment:
This change makes the code worse IMO. The null check was nicely hidden in the util method, and now it pops out.